### ISSUE TITLE
Split `official.yml` into multiple jobs

### ISSUE THREAD

**Issue by MiYanni** [dotnet/project-system#7914]
While doing the OptProf work, I noticed that our `official.yml` is basically split into sections. I ended up adding comment headers to each section. These sections, instead of being done in-series, can be done in-parallel. The best way to manage this is to make a separate `.yml` per **job**.

A job can be ran in parallel of other jobs. They can multiple steps like a normal pipeline. Technically, a normal pipeline is simply a *single job pipeline*. I've done this before in [azure-powershell](https://github.com/Azure/azure-powershell/blob/main/.azure-pipelines/powershell-core.yml) where the `Build` job runs, and then the `Analyze` and `Test` jobs depend on the build succeeding, but run in parallel of each other. I can do something similar here were multiple things can happen simultaneously after the build succeeds. This also means we don't have to have 1990's-style section headers in our `.yml` file. 🤣

For us, we do a lot of publishing. Right now, we have no independent testing (meaning, external to the build) done in our official pipeline. If we add integration tests, it would be good to add those as a separate job. One benefit too is that we could no longer use our release pipelines, but instead, make the insertion PR creation a job that activates as part of our official build. I'll make that endeavor as a separate issue.

## Sections
https://github.com/dotnet/project-system/blob/c8137374c0f7e6458397627b530ddb14099cb6b9/build/ci/official.yml#L37-L39
https://github.com/dotnet/project-system/blob/c8137374c0f7e6458397627b530ddb14099cb6b9/build/ci/official.yml#L68-L70
https://github.com/dotnet/project-system/blob/c8137374c0f7e6458397627b530ddb14099cb6b9/build/ci/official.yml#L104-L106
https://github.com/dotnet/project-system/blob/c8137374c0f7e6458397627b530ddb14099cb6b9/build/ci/official.yml#L150-L152
https://github.com/dotnet/project-system/blob/c8137374c0f7e6458397627b530ddb14099cb6b9/build/ci/official.yml#L205-L207
https://github.com/dotnet/project-system/blob/c8137374c0f7e6458397627b530ddb14099cb6b9/build/ci/official.yml#L241-L243

**Comment by MiYanni** [comment]
This relates to: https://github.com/dotnet/project-system/issues/7915
